mpact/services.py

```diff
@@ -135,16 +141,21 @@ async def validate_code(client, data):
     return {
         DATA: {
             FIRST_NAME: user_details.first_name,
-            LAST_NAME: user_details.last_name,
-            USERNAME: user_details.username,
+            TOKEN: encode_token(
```

Comment:
You could probably move this whole return value into a function so that you don't repeat yourself. e.g.
```python
    # ...
    return ok_response(user_details, data[PHONE])


def ok_response(user_details, phone):
    return {
        DATA: {
            # ...
```

Reply:
How did you choose which user details to include in the token, and which to pass in plaintext? Should first name be included? Should username not be included? :shrug:

Reply:
I chose based on the details which are being used on the UI side. Firstname is used to display after signing in that's why I kept it in plain text. I'm not displaying the username because the username is not required by the telegram while creating an account so it may be empty.